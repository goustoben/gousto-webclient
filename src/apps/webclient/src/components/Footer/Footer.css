@value sizeLG, sizeXXL from 'styles/typography.css';
@value SpacingXXS, SpacingSM from 'sass-loader!styles/spacing.scss';
@value Blackberry, White from 'styles/colors.css';
@value MaxBreakPointXS, ScreenXSMax, ScreenSMMin, ScreenXXSMax, ScreenSMMax, ScreenMLMin, ScreenLGMin from 'styles/screenSizes.css';
@value BreakpointFromMedium from 'design-language/breakpoints.css';


.superContainer {
	composes: container-fluid from 'sass-loader!styles/bootstrap.scss';
}

.nav {
	composes: container from 'sass-loader!styles/bootstrap.scss';
	text-align: center;
	/* if IE 11 */
	@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none), (max-width: MaxBreakPointXS) {
		flex: 0 0 100%;
	}
}

.links {
	composes: noPadding noMargin from 'sass-loader!styles/spacing.scss';
	padding-bottom: 20px;
	list-style: none;
}

.link {
	composes: textSM from 'styles/typography.css';
	composes: inlineBlock from 'styles/display.css';
}

.icon {
	composes: marginXSLeftRight from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	margin-top: calc(SpacingXXS * -1);
	vertical-align: top;
	color: inherit;
}

.patternSvg {
	background-size: contain;
	background-repeat: repeat-x;
	width: 100%;
	min-height: 500px;
}

.phone {
	composes: textSM from 'styles/typography.css';
	@media ScreenXSMax {
		line-height: 3;
	}
}
.phone, .phone:hover, .phone:active {
	color: White !important;
	text-decoration: none;
	position: relative;
	padding-left: 23px;
}

.earPhone {
	composes: fa-basic-class fa-phone from 'styles/vendor/font-awesome-module.css';
	composes: paddingXSRight from 'sass-loader!styles/spacing.scss';
	font-size: sizeLG;
	position: absolute;
	top: 0px;
	left: 0px;
}

.pipe {
	composes: paddingXSTopBottom paddingXSLeftRight marginLGRight marginXXSLeft from 'sass-loader!styles/spacing.scss';
	border-right: 1px solid White;
}

.mobilePipe {
	composes: pipe;
	height: 27px;
}

.desktopHide {
	composes: hidden-md-up from 'sass-loader!styles/bootstrap.scss';
}

.desktopShow {
	composes: hidden-md-down from 'sass-loader!styles/bootstrap.scss';
}

.mobileFooter {
	composes: row col-xs-3 center-block from 'sass-loader!styles/bootstrap.scss';
	composes: desktopHide;
}

.newsletterContainer {
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
	composes: marginSMTopBottom from 'sass-loader!styles/spacing.scss';
}

.menu {
	composes: row from 'sass-loader!styles/bootstrap.scss';
}

.menuList {
	list-style: none;
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
	padding-bottom: 25px;
	padding-top: 20px;

	@media ScreenMLMin {
		padding-top: 40px;
	}
}

.menuList li {
	@media ScreenLGMin {
		margin-left: 10px;
		margin-right: 10px;
	}
}

.menuItem {
	composes: inlineBlock from 'styles/display.css';
	composes: sizeMD axiforma from 'styles/typography.css';
	cursor: pointer;
	@media ScreenSMMax {
		font-weight: normal;
	}

	&:not(:last-child):after{
		 content: '';
		 padding: SpacingSM;
		 padding-top: 0;
		 padding-bottom: 0;
		 @media ScreenSMMax {content: '|';}
	 }
}

.separator {
	composes: desktopHide;
	width: 100%;
	height: 1px;
	border: none;
	background: White;
	opacity: 0.4;
}

.copyright {
	composes: paragraphXS from 'styles/typography.css';
	composes: textXXS from 'styles/typography.css';
	padding-bottom: 5px;
}

.contact {
	composes: marginMDTop from 'sass-loader!styles/spacing.scss';
}

.appStoreLinksContainer {
	composes: marginMDBottom from 'sass-loader!styles/spacing.scss';
}

.contact, .appStoreLinksContainer {
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
}

.row {
	composes: row from 'sass-loader!styles/bootstrap.scss';
}

.footer {
	composes: textCenter from 'styles/typography.css';
	background: White;
	padding-bottom: 0;
}

.appLinks {
	composes: paddingXXXXLTopBottom from 'sass-loader!styles/spacing.scss';
	padding-bottom: 15px;
}

.icon {
	composes: marginXSLeftRight from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	margin-top: calc(SpacingXXS * -1);
	vertical-align: top;
	font-size: sizeXXL;
}

.facebook{
	composes: icon;
	composes: fa-basic-class fa-facebook from 'styles/vendor/font-awesome-module.css';
}

.twitter{
	composes: icon;
	composes: fa-basic-class fa-twitter from 'styles/vendor/font-awesome-module.css';
}

.youtube{
	composes: icon;
	composes: fa-basic-class fa-youtube from 'styles/vendor/font-awesome-module.css';
}

.instagram{
	composes: icon;
	composes: fa-basic-class fa-instagram from 'styles/vendor/font-awesome-module.css';
}

.modernSlaveryLink {
	@media ScreenXSMax {
		display: none;
	}
}

.numberOneNotice {
  padding: 0 0.75rem 1rem;
  margin: auto;

  @media BreakpointFromMedium {
    padding: 0 2rem 1rem;
    max-width: 60rem;
  }
}
