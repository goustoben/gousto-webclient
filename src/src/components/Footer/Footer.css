@value sizeXS, sizeLG, sizeXL, sizeXXL from 'styles/typography.css';
@value SpacingXXS, SpacingXS, SpacingSM, SpacingMD, SpacingLG, SpacingXL, SpacingXXL, SpacingXXXL from 'sass-loader!styles/spacing.scss';
@value Black, Blackberry, Cacao, Chia, Mushroom50, Spruce, White from 'styles/colors.css';
@value MaxBreakPointXS from 'styles/screenSizes.css';
@value ScreenXSMax, ScreenSMMin, ScreenXXSMax, ScreenSMMax, ScreenMDMin, ScreenMLMin, ScreenLGMin, ScreenMDMax from 'styles/screenSizes.css';

.superContainer {
	composes: container-fluid from 'sass-loader!styles/bootstrap.scss';
}

.nav {
	composes: container from 'sass-loader!styles/bootstrap.scss';
	text-align: center;
	/* if IE 11 */
	@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none), (max-width: MaxBreakPointXS) {
		flex: 0 0 100%;
	}
}

.links {
	composes: noPadding noMargin from 'sass-loader!styles/spacing.scss';
	padding-bottom: 20px;
	list-style: none;
}

.link {
	composes: textSM from 'styles/typography.css';
	composes: inlineBlock from 'styles/display.css';
}

.icon {
	composes: marginXSLeftRight from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	margin-top: calc(SpacingXXS * -1);
	vertical-align: top;
	color: inherit;
}

.patternSvg {
	background-size: contain;
	background-repeat: repeat-x;
	width: 100%;
	min-height: 500px;
}

.phone {
	composes: textSM from 'styles/typography.css';
	@media ScreenXSMax {
		line-height: 3;
	}
}
.phone, .phone:hover, .phone:active {
	color: White !important;
	text-decoration: none;
	position: relative;
	padding-left: 23px;
}

.earPhone {
	composes: fa-basic-class fa-phone from 'styles/vendor/font-awesome-module.css';
	composes: paddingXSRight from 'sass-loader!styles/spacing.scss';
	font-size: sizeLG;
	position: absolute;
	top: 0px;
	left: 0px;
}

.pipe {
	composes: paddingXSTopBottom paddingXSLeftRight marginLGRight marginXXSLeft from 'sass-loader!styles/spacing.scss';
	border-right: 1px solid White;
}

.mobilePipe {
	composes: pipe;
	height: 27px;
}

.mobileHide {
	composes: hidden-sm-down from 'sass-loader!styles/bootstrap.scss';
}

.desktopHide {
	composes: hidden-md-up from 'sass-loader!styles/bootstrap.scss';
}

.desktopShow {
	composes: hidden-md-down from 'sass-loader!styles/bootstrap.scss';
}

.mobileFooter {
	composes: row col-xs-3 center-block from 'sass-loader!styles/bootstrap.scss';
	composes: desktopHide;
}

.newsletterContainer {
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
	composes: marginSMTopBottom from 'sass-loader!styles/spacing.scss';
}

.menu {
	composes: row from 'sass-loader!styles/bootstrap.scss';
}

.menuList {
	list-style: none;
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
	padding-bottom: 25px;
	padding-top: 20px;

	@media ScreenMLMin {
		padding-top: 40px;
	}
}

.menuList li {
	@media ScreenLGMin {
		margin-left: 10px;
		margin-right: 10px;
	}
}

.menuItem {
	composes: inlineBlock from 'styles/display.css';
	composes: sizeMD axiforma from 'styles/typography.css';
	@media ScreenSMMax {
		font-weight: normal;
	}

	&:not(:last-child):after{
		 content: '';
		 padding: SpacingSM;
		 padding-top: 0;
		 padding-bottom: 0;
		 @media ScreenSMMax {content: '|';}
	 }
}

.separator {
	composes: desktopHide;
	width: 100%;
	height: 1px;
	border: none;
	background: White;
	opacity: 0.4;
}

.copyright {
	composes: paragraphXS from 'styles/typography.css';
	composes: textXXS from 'styles/typography.css';
	padding-bottom: 5px;
	/*color: White;*/
}

.contact {
	composes: marginMDTop from 'sass-loader!styles/spacing.scss';
}

.appStoreLinksContainer {
	composes: marginMDBottom from 'sass-loader!styles/spacing.scss';
}

.contact, .appStoreLinksContainer {
	composes: col-xs-12 from 'sass-loader!styles/bootstrap.scss';
}

.row {
	composes: row from 'sass-loader!styles/bootstrap.scss';
}

.footer {
	composes: textCenter from 'styles/typography.css';
	background: White;
	padding-bottom: 0;
}

.appLinks {
	composes: paddingXXXXLTopBottom from 'sass-loader!styles/spacing.scss';
	padding-bottom: 15px;
}

.icon {
	composes: marginXSLeftRight from 'sass-loader!styles/spacing.scss';
	composes: inlineBlock from 'styles/display.css';
	margin-top: calc(SpacingXXS * -1);
	vertical-align: top;
	font-size: sizeXXL;
}

.facebook{
	composes: icon;
	composes: fa-basic-class fa-facebook from 'styles/vendor/font-awesome-module.css';
}

.twitter{
	composes: icon;
	composes: fa-basic-class fa-twitter from 'styles/vendor/font-awesome-module.css';
}

.youtube{
	composes: icon;
	composes: fa-basic-class fa-youtube from 'styles/vendor/font-awesome-module.css';
}

.instagram{
	composes: icon;
	composes: fa-basic-class fa-instagram from 'styles/vendor/font-awesome-module.css';
}

.checkoutFooterContainer {
	composes: paddingMDTop from 'sass-loader!styles/spacing.scss';
	margin: auto;
	width: 100%;
	min-height: 100%;
	background: Blackberry;
	height: 175px;
	color: White;
	display: flex;
	justify-content: center;
	align-items: center;

	@media ScreenSMMin {
		margin-top: -156px;
	}

	@media ScreenXSMax {
		margin-top: -90px;
	}
}

.footerFlexContainer {
	display: flex;
	align-items: center;
	justify-content: center;
	height: 100%;
	flex-wrap: wrap;
	max-width: 700px;

	@media ScreenXXSMax {
		flex-direction: column;
		align-items: center;
		max-width: 100%;
		width: 100%;
		margin-right: 0px;
	}
}

.greatValue {
	display: flex;
	justify-content: center;
	max-width: 230px;
	flex-wrap: wrap;
	margin-right: 60px;
	max-width: auto;
	width: auto;

	@media ScreenXXSMax {
		flex-direction: column;
		align-items: center;
		width: 100%;
		margin-right: 0px;
	}

	@media ScreenXSMax {
		max-width: 158px;
	}
}

.greatValueTitle {
	margin: 0;
	margin-right: 10px;
}

.greatValueDescription {
	margin-top: 10px;
	font-size: 0.8rem;
}

.trustpilotContainer {
	width: auto;

	@media ScreenXXSMax {
		width: 100%;
		margin-bottom: 10px;
		margin-top: 10px;
	}
}

.iconTrustpilot {
	height: 30px;
	width: 200px;

	@media ScreenSMMin {
		margin-top: -15px;
	}

	@media ScreenXSMax {
		height: 30px;
		width: 130px;
	}
}

.icon5Stars {
	height: 29px;
	width: 150px;

	@media ScreenXSMax {
		height: 20px;
		width: 100px;
	}
}

.copyrightCheckout {
	margin-top: -15px;
	width: 100%;
	font-size: 0.9rem;

	@media ScreenXSMax {
		font-size: 0.7rem;
	}
}

.modernSlaveryLink {
	@media ScreenXSMax {
		display: none;
	}
}
