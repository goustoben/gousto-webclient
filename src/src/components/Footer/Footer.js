import PropTypes from 'prop-types'
import React from 'react'
import moment from 'moment'
import config from 'config'
import * as trackingKeys from 'actions/trackingKeys'
import { client as clientRoutes } from 'config/routes'
import { trustPilotReviews } from 'config/home'
import classNames from 'classnames'
import { onEnter } from 'utils/accessibility'
import { AppStoreLinks } from 'components/AppStoreLinks'
import Svg from 'components/Svg'
import Link from 'Link'
import css from './Footer.css'

const showHelpPreLogin = (helpPreLoginVisibilityChange) => (
  helpPreLoginVisibilityChange(true)
)

const renderHelpLink = (
  isAuthenticated,
  helpPreLoginVisibilityChange,
  isHelpCentreActive,
  trackNavigationClick
) => {
  const getHelpRoute = clientRoutes.getHelp

  return (
    isAuthenticated
      ? (
        <Link
          to={isHelpCentreActive
            ? `${clientRoutes.helpCentre}`
            : `${getHelpRoute.index}/${getHelpRoute.eligibilityCheck}`}
          data-optimizely="footer-help-link"
          data-selid="footer-learn-more"
          title="Help"
          clientRouted={false}
          secondary
          tracking={() => trackNavigationClick({
            actionType: trackingKeys.clickHelpFooter,
            seCategory: 'help',
            logged_in: true,
          })}
        >
          Help
        </Link>
      )
      : (
        <span
          data-test="help-link"
          role="button"
          tabIndex="0"
          onClick={() => {
            showHelpPreLogin(helpPreLoginVisibilityChange)
            trackNavigationClick({
              actionType: trackingKeys.clickHelpFooter,
              seCategory: 'help',
              logged_in: false,
            })
          }}
          onKeyDown={onEnter(() => showHelpPreLogin(helpPreLoginVisibilityChange))}
        >
          Help
        </span>
      )
  )
}

const Footer = ({
  copyright,
  helpPreLoginVisibilityChange,
  isAuthenticated,
  simple,
  type,
  isHelpCentreActive,
  trackNavigationClick,
  isMenuRedirectPageEnabled,
  postCode,
}) => {
  const renderTermsLink = () => (
    <li className={css.menuItem}>
      Terms
      (
      <Link to={clientRoutes.termsOfUse} data-selid="footer-terms-of-use" title="Terms &amp; Conditions" clientRouted={false} secondary>Website</Link>
      )
      (
      <Link to={clientRoutes.termsAndConditions} data-selid="footer-terms-and-conditions" title="Terms &amp; Conditions for Sale of Goods" clientRouted={false} secondary>Sale</Link>
      )
    </li>
  )

  const renderPrivacyLink = () => (
    <li className={css.menuItem}>
      <Link to={clientRoutes.privacyPolicy} data-selid="footer-privacy-statement" title="Privacy Policy" clientRouted secondary>Privacy Policy</Link>
    </li>
  )

  const renderModernSlaveryLink = () => (
    <li className={css.menuItem}>
      <Link to={clientRoutes.modernSlavery} data-selid="footer-modern-slavery-statement" title="Modern Slavery" clientRouted secondary>
        Modern Slavery
        <span className={css.modernSlaveryLink}> Statement</span>
      </Link>
    </li>
  )

  const renderSimpleList = () => (
    <ul className={css.menuList}>
      {renderTermsLink()}
      {renderPrivacyLink()}
      {renderModernSlaveryLink()}
    </ul>
  )

  const isRedirectToNewMenu = !isAuthenticated && !postCode && isMenuRedirectPageEnabled
  const trackWeeklyRecipesClick = () => trackNavigationClick({ actionType: trackingKeys.clickRecipeNavigationFooter })
  const renderFullList = () => (
    <ul className={css.menuList}>
      <li className={classNames(css.mobileHide, css.menuItem)}>
        <Link to={clientRoutes.home} data-selid="footer-home" title="Home" clientRouted={false} secondary>Home</Link>
      </li>
      <li className={classNames(css.mobileHide, css.menuItem)}>
        {/* eslint-disable-next-line */}
        <span data-test="week-recipes" onClick={trackWeeklyRecipesClick} role="button" tabIndex={0}>
          <Link
            to={isRedirectToNewMenu ? clientRoutes.menu2 : clientRoutes.menu}
            data-selid="footer-this-weeks-recipes"
            title="This Week's Recipes"
            clientRouted
            secondary
          >
            This Week&apos;s Recipes
          </Link>
        </span>
      </li>
      <li className={css.menuItem}>
        {renderHelpLink(isAuthenticated, helpPreLoginVisibilityChange, isHelpCentreActive, trackNavigationClick)}
      </li>
      {renderTermsLink()}
      <li className={css.menuItem}>
        <Link data-selid="footer-cookbook" to={clientRoutes.cookbook} title="Cookbook" clientRouted={false} secondary>Cookbook</Link>
      </li>
      <li className={css.menuItem}>
        <Link to={clientRoutes.jobs} data-selid="footer-jobs" title="Jobs" clientRouted={false} secondary>Jobs</Link>
      </li>
      <li className={css.menuItem}>
        <Link to={clientRoutes.weCare} data-selid="footer-we-care" title="We Care" clientRouted={false} secondary>We Care</Link>
      </li>
      <li className={css.menuItem}>
        <Link to={clientRoutes.blog} data-selid="footer-blog" title="Blog" clientRouted={false} secondary>Blog</Link>
      </li>
      <li className={classNames(css.mobileHide, css.menuItem)}>
        <Link to={clientRoutes.ourSuppliers} data-selid="footer-our-suppliers" title="Our Suppliers" clientRouted={false} secondary>Our Suppliers</Link>
      </li>
      {renderPrivacyLink()}
      {renderModernSlaveryLink()}
    </ul>
  )

  const renderSocial = () => (
    <div className={css.appLinks}>
      <ul className={css.links}>
        <li className={css.link}>
          <Link to="http://www.facebook.com/goustocooking" data-selid="footer-facebook" clientRouted={false} secondary>
            <span role="img" aria-label="facebook" className={css.facebook} />
          </Link>
        </li>
        <li className={css.link}>
          <Link to="https://twitter.com/goustocooking" data-selid="footer-twitter" clientRouted={false} secondary>
            <span role="img" aria-label="twitter" className={css.twitter} />
          </Link>
        </li>
        <li className={css.link}>
          <Link to="https://www.youtube.com/UKGousto/" data-selid="footer-youtube" clientRouted={false} secondary>
            <span role="img" aria-label="youtube" className={css.youtube} />
          </Link>
        </li>
        <li className={css.link}>
          <Link to="http://instagram.com/goustocooking" data-selid="footer-instagram" clientRouted={false} secondary>
            <span role="img" aria-label="instagram" className={css.instagram} />
          </Link>
        </li>
      </ul>
      <AppStoreLinks appStoreId={config.apps.appStoreId} playStoreId={config.apps.playStoreId} />
    </div>
  )

  const copyrightText = (
    <p>
      &copy; Gousto
      {moment().format('YYYY')}
      . All rights reserved.
    </p>
  )

  const renderCopyright = () => (
    <div id="copyright" className={css.copyright}>
      {copyrightText}
    </div>
  )

  const renderSimpleFooter = () => (
    <div>
      {renderSimpleList()}
      {renderCopyright()}
    </div>
  )

  const renderMediumFooter = () => (
    <div>
      {renderSocial()}
      {renderFullList()}
      {copyright && renderCopyright()}
    </div>
  )

  const renderLargeFooter = () => (
    <div>
      {renderMediumFooter()}
    </div>
  )

  const renderCheckoutFooter = () => (
    <div className={css.checkoutFooterContainer}>
      <div className={css.footerFlexContainer}>
        <div className={css.greatValue}>
          <h3 className={css.greatValueTitle}>9 / 10</h3>
          <Svg fileName="icon-5-stars-trustpilot" className={css.icon5Stars} />
          <div className={css.greatValueDescription}>
            {`Based on ${trustPilotReviews}+ reviews`}
          </div>
        </div>
        <div className={css.trustpilotContainer}>
          <Svg fileName="icon-trustpilot-logo-darkbg" className={css.iconTrustpilot} />
        </div>
        <div className={css.copyrightCheckout}>
          {copyrightText}
        </div>
      </div>
    </div>
  )

  const renderFooter = () => {
    let footer = null

    if (simple) {
      // Back compatibility of simple prop
      return renderSimpleFooter()
    }

    switch (type) {
    case 'simple':
      footer = renderSimpleFooter()
      break
    case 'medium':
      footer = renderMediumFooter()
      break
    case 'large':
      footer = renderLargeFooter()
      break
    case 'checkout':
      footer = renderCheckoutFooter()
      break
    default: {
      footer = renderMediumFooter()
      break
    }
    }

    return footer
  }

  return (
    <footer className={css.footer}>
      {renderFooter()}
    </footer>
  )
}

Footer.propTypes = {
  copyright: PropTypes.bool,
  isAuthenticated: PropTypes.bool,
  helpPreLoginVisibilityChange: PropTypes.func.isRequired,
  simple: PropTypes.bool,
  type: PropTypes.string,
  isHelpCentreActive: PropTypes.bool,
  trackNavigationClick: PropTypes.func.isRequired,
  isMenuRedirectPageEnabled: PropTypes.bool,
  postCode: PropTypes.string,
}

Footer.defaultProps = {
  copyright: true,
  isAuthenticated: false,
  simple: false,
  type: 'medium',
  isHelpCentreActive: false,
  isMenuRedirectPageEnabled: false,
  postCode: '',
}

// eslint-disable-next-line
export default Footer
