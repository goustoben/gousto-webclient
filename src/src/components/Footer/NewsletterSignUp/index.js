export { default as Component } from './NewsletterSignUpContainer'
