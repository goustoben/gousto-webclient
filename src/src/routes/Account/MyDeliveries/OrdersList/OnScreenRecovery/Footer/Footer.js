import PropTypes from 'prop-types'
import React from 'react'

import { ModalFooter } from 'ModalComponent'

import css from './Footer.css'

const propTypes = {
  onConfirm: PropTypes.func.isRequired,
  confirmCopy: PropTypes.string.isRequired,
  onKeep: PropTypes.func.isRequired,
  keepCopy: PropTypes.string.isRequired,
}

const Footer = ({ onConfirm, confirmCopy, onKeep, keepCopy }) => {

  return (
    <ModalFooter>
      <div
        className={css.loss}
        role="button"
        tabIndex={0}
        onClick={() => onConfirm()}
        onKeyDown={event => event.keyCode === 13 && onConfirm()}
      >
        {confirmCopy}
      </div>
      <button
        className={css.recover}
        type="button"
        onClick={() => onKeep()}
        onKeyDown={event => event.keyCode === 13 && onKeep()}
      >
        {keepCopy}
      </button>
    </ModalFooter>
  )
}

Footer.propTypes = propTypes

export { Footer }
