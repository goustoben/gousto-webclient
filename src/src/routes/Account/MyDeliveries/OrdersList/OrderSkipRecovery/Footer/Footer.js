import React, { PropTypes } from 'react'

import { ModalFooter } from 'ModalComponent'

import css from './Footer.css'

const propTypes = {
    callToActions: PropTypes.shape({
		confirm: PropTypes.string,
		keep: PropTypes.string,
	}),
    onClickKeepOrder: PropTypes.func,
    onClickSkipCancel: PropTypes.func,
}

const Footer = ({ orderType, callToActions, onClickKeepOrder, onClickSkipCancel }) => {
    let keepCopy
    let confirmCopy

    if (callToActions) {
        keepCopy = callToActions.keep
        confirmCopy = callToActions.confirm
    } else {
        keepCopy = 'Keep Box'
        confirmCopy = `${orderType === 'pending' ? 'Cancel' : 'Skip'} anyway`
    }

    return (
        <ModalFooter>
            <button className={css.keepButton} onClick={() => onClickKeepOrder}>
                {keepCopy}
            </button>
            <div className={css.skipAnyWay} onClick={() => onClickSkipCancel}>
                {confirmCopy}
            </div>
        </ModalFooter>
    )
}

Footer.propTypes = propTypes

export default Footer
