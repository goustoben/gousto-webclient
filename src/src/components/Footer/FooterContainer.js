import { connect } from 'react-redux'
import { getIsAuthenticated } from 'selectors/auth'
import Footer from './Footer'
import { helpPreLoginVisibilityChange } from "actions/login/helpPreLoginVisibilityChange"
import { trackNavigationClick } from "actions/tracking/trackNavigationClick"

const mapStateToProps = (state, ownProps) => ({
  isAuthenticated: getIsAuthenticated(state),
  simple: ownProps.simple || state.persist.get('simpleHeader', false),
  type: ownProps.type,
})

export default connect(mapStateToProps, {
  helpPreLoginVisibilityChange,
  trackNavigationClick,
})(Footer)
