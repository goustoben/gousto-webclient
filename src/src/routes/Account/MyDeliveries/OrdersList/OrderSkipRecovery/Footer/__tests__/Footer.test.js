import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Footer from '../Footer'

describe('Order Skip Recovery Footer', () => {
    const mockKeepOrder = jest.fn()
    const mockSkipOrder = jest.fn()
    let callToActions

    beforeEach(() => {
        callToActions = {
            keep: 'keep',
            confirm: 'confirm',
        }
    })
    describe('Initial Render', () => {
        let wrapper

        beforeAll(() => {
            wrapper = mount(
                <Footer orderType="pending" callToActions={callToActions} onClickKeepOrder={mockKeepOrder} onClickSkipCancel={mockSkipOrder} />
            )
        })

        test('should render snapshot', () => {
            const tree = renderer.create(
                <Footer onClickKeepOrder={mockKeepOrder} onClickSkipCancel={mockSkipOrder} />
            ).toJSON()

			expect(tree).toMatchSnapshot()
        })

        test('should display keep button', () => {
            const keepBtn = wrapper.find('button.keepButton')

            expect(keepBtn.length).toBe(1)
            expect(keepBtn.text()).toBe('Keep Box')
        })

        test('should display skip button', () => {
            const skipBtn = wrapper.find('div.skipAnyWay')

            expect(skipBtn.length).toBe(1)
            expect(skipBtn.text()).toBe('Cancel anyway')
        })
    })

    describe('Alternative Render', () => {

    })

    describe.skip('Functionality', () => {
        let wrapper

        beforeAll(() => {
            wrapper = mount(
                <Footer orderType="pending" callToActions={callToActions} onClickKeepOrder={mockKeepOrder} onClickSkipCancel={mockSkipOrder} />
            )
        })

        test('should fire keep order click event', () => {
            const keepBtn = wrapper.find('button.keepButton')

            keepBtn.simulate('click')

            expect(mockKeepOrder).toHaveBeenCalledTimes(1)
        })

        test('should fire skip order click event', () => {
            const skipBtn = wrapper.find('div.skipAnyWay')

            skipBtn.simulate('click')

            expect(mockSkipOrder).toHaveBeenCalledTimes(1)
        })
    })
})
