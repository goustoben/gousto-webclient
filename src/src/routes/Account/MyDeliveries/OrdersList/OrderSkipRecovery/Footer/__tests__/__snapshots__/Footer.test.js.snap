// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Recovery Footer Initial Render should render snapshot 1`] = `
<div
  className="modalFooter"
>
  <button
    className="keepButton"
    onClick={[Function]}
  >
    Keep Box
  </button>
  <div
    className="skipAnyWay"
    onClick={[Function]}
  >
    Skip anyway
  </div>
</div>
`;
